docs/docs-components/data/types.js

```diff
@@ -12,3 +12,64 @@ export type DesignOverview = {|
   path: string,
   svg: SVGElement,
 |};
+
+type StatusType = 'notAvailable' | 'partial' | 'planned' | 'ready';
+
+/**
+ * All components should be in a single list. That list can be filtered by category/platform/etc where needed.
+ */
+
+type ComponentAccessibility = {|
+  a11yComprehension?: StatusType,
+  a11yNavigation?: StatusType,
+  a11yScreenreader?: StatusType,
+  a11yVisual?: StatusType,
+  summary?: StatusType,
+|};
+
+type ComponentCategory =
+  | 'Actions'
+  | 'Avatars'
+  | 'Building blocks'
+  | 'Controls'
+  | 'Data'
+  | 'Fields and forms'
+  | 'Help and guidance'
+  | 'Indicators'
+  | 'Loading'
+  | 'Messaging'
+  | 'Navigation'
+  | 'Overlays'
+  | 'Pins and imagery'
+  | 'Structure'
+  | 'Text'
+  | 'Utilities';
+
+type PlatformObjType<T> = {|
+  android?: T,
+  figma?: T,
+  ios?: T,
+  web?: T,
+|};
+
+export type ComponentData = {|
+  id: string, // This is essentially the internal name
```

Comment:
Since components might have different names by platform, this gives us a single "name" to refer to the component by when reference component data objects.